google/auth/jwt.py

```diff
@@ -426,13 +426,15 @@ def with_claims(self, issuer=None, subject=None, audience=None,
         Returns:
             google.auth.jwt.Credentials: A new credentials instance.
         """
+        new_additional_claims = self._additional_claims.copy()
```

Comment:
Nah, not horribly concerned about deep nesting, it's pretty rare to do that in the jwt payload.